### PR TITLE
fix migrate-olap to use tenant-aware location for staging files on S3

### PR DESCRIPTION
**Pull request by malaffoon**
I tried using SQL params and a couple other approaches but none of them worked. So, brute force time ...

### CODE REVIEW

File: migrate-olap/src/main/java/org/opentestsystem/rdw/ingest/migrate/olap/MigrateOlapReportingApplication.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
 @PropertySource(value = "classpath:build-info.properties", ignoreResourceNotFound = true)
 public class MigrateOlapReportingApplication {
 
-    public static void main(final String[] args) throws Exception {
+    public static void main(final String[] args) {
         SpringApplication.run(MigrateOlapReportingApplication.class, args);
     }
-}
\ No newline at end of file
+}
```
Comment (malaffoon) [inline migrate-olap/src/main/java/org/opentestsystem/rdw/ingest/migrate/olap/MigrateOlapReportingApplication.java]: unrelated, fix warning

File: migrate-olap/src/main/java/org/opentestsystem/rdw/ingest/migrate/olap/repository/SqlListExecutionWithMigrateRepository.java [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 
 
+import org.opentestsystem.rdw.ingest.common.repository.SqlListExecutionRepository;
 import org.opentestsystem.rdw.migrate.common.Migrate;
 
 /**
  * Repository for executing a collection of the sqls that require {@link Migrate} attributes as parameters
  */
-public interface SqlListExecutionWithMigrateRepository {
+public interface SqlListExecutionWithMigrateRepository extends SqlListExecutionRepository {
```
Comment (malaffoon) [inline migrate-olap/src/main/java/org/opentestsystem/rdw/ingest/migrate/olap/repository/SqlListExecutionWithMigrateRepository.java]: Strictly speaking not necessary but it avoids having to have two repositories in the config class (might make more sense when you get to that file).

File: migrate-olap/src/main/java/org/opentestsystem/rdw/ingest/migrate/olap/repository/impl/JdbcSqlListPlaceholderExecutionRepository.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+     * @param jdbcTemplate the warehouse jdbc template
+     * @param archiveService used to determine migration location placeholder value
+     */
+    protected JdbcSqlListPlaceholderExecutionRepository(final NamedParameterJdbcTemplate jdbcTemplate,
+                                                        final ArchiveService archiveService) {
+        super(jdbcTemplate);
+        this.archiveService = archiveService;
+    }
+
+    @Override
```
Comment (malaffoon) [inline migrate-olap/src/main/java/org/opentestsystem/rdw/ingest/migrate/olap/repository/impl/JdbcSqlListPlaceholderExecutionRepository.java]: This PR boils down to this method: using the dynamic archive service, figure out what the appropriate S3 location is for staging the files and stuff that into the SQL statements.

File: migrate-olap/src/main/java/org/opentestsystem/rdw/ingest/migrate/olap/repository/impl/JdbcSqlListPlaceholderExecutionRepository.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    protected Iterable<String> checkSql(final Iterable<String> sqls) {
+        final Properties properties = new Properties();
+        // this should be something like "s3://rdw-ci-archive/CA/MigrateOlap"
+        properties.put("migrateLocationPlaceholder", archiveService.getRawUri(location));
+
+        final List<String> replacedSqls = newArrayList();
+        for (final String sql : sqls) {
+            replacedSqls.add(placeholderHelper.replacePlaceholders(sql, properties));
+        }
+        return replacedSqls;
```
Comment (malaffoon) [inline migrate-olap/src/main/java/org/opentestsystem/rdw/ingest/migrate/olap/repository/impl/JdbcSqlListPlaceholderExecutionRepository.java]: Anybody else find it annoying that `Iterable` can't be easily turned into a stream?  Am i missing something?

File: migrate-olap/src/main/java/org/opentestsystem/rdw/ingest/migrate/olap/step/CleanupLandingLocation.java [hunk 5]
```diff
@@ -37,4 +37,4 @@ private void deleteLocationIfExists() {
             //location does not exists
         }
     }
-}
\ No newline at end of file
+}
```
Comment (malaffoon) [inline migrate-olap/src/main/java/org/opentestsystem/rdw/ingest/migrate/olap/step/CleanupLandingLocation.java]: unrelated, fix warning

File: migrate-olap/src/main/java/org/opentestsystem/rdw/ingest/migrate/olap/step/OlapReportingMigrateStepsConfig.java [hunk 6]
```diff
@@ -16,18 +9,29 @@
...
+ * with the $[migrateLocationPlaceholder] substitution.
+ * </p>
  */
 @Configuration
 public class OlapReportingMigrateStepsConfig {
     private final StepBuilderFactory stepBuilderFactory;
 
-    private final SqlListExecutionWithMigrateRepository warehouseCopyRepository;
-    private final SqlListExecutionRepository warehouseSqlListExecutionRepository;
+    private final SqlListExecutionWithMigrateRepository warehouseSqlListExecutionRepository;
```
Comment (malaffoon) [inline migrate-olap/src/main/java/org/opentestsystem/rdw/ingest/migrate/olap/step/OlapReportingMigrateStepsConfig.java]: That interface extends thing up above let's me get rid of one of these.

File: migrate-olap/src/main/java/org/opentestsystem/rdw/ingest/migrate/olap/step/SqlListEmbargoExecutionStep.java [hunk 7]
```diff
@@ -29,10 +29,10 @@
...
-    public RepeatStatus execute(final StepContribution contribution, final ChunkContext chunkContext) throws Exception {
+    public RepeatStatus execute(final StepContribution contribution, final ChunkContext chunkContext) {
         if (getJobBatch(chunkContext).isMigrateEmbargo()) {
             repository.execute(sqls);
         }
         return RepeatStatus.FINISHED;
     }
-}
\ No newline at end of file
+}
```
Comment (malaffoon) [inline migrate-olap/src/main/java/org/opentestsystem/rdw/ingest/migrate/olap/step/SqlListEmbargoExecutionStep.java]: more warning fixes

File: migrate-olap/src/main/java/org/opentestsystem/rdw/ingest/migrate/olap/repository/impl/JdbcSqlListPlaceholderExecutionRepository.java (hunk 4 above)
Comment (ftdholbrook) [reply migrate-olap/src/main/java/org/opentestsystem/rdw/ingest/migrate/olap/repository/impl/JdbcSqlListPlaceholderExecutionRepository.java]: StreamSupport.stream(iterable.spliterator(), false);

Comment (malaffoon) [reply migrate-olap/src/main/java/org/opentestsystem/rdw/ingest/migrate/olap/repository/impl/JdbcSqlListPlaceholderExecutionRepository.java]: Yeah, i guess it depends on your definition of easily  ;)

**Pull request merged**